src/TextDateInput/TextDateInput.tsx

```diff
@@ -0,0 +1,121 @@
+import React from 'react'
+import styled from 'styled-components'
+
+import { TextInput } from '../TextInput'
+import { Box } from '../Box'
+import { Fieldset, FieldsetProps } from '../fields/Fieldset'
+
+export type FullDateFormat = {
+  day: string | null
+  month: string | null
+  year: string | null
+}
+
+type ValidationErrorType =
+  | 'day'
+  | 'month'
+  | 'year'
+  | 'all'
```

Comment:
I think there will be more errors types encopassed under `all`. Maybe a better approach would be to pass an array as a prop for example, fieldsWithError: ['day', 'month'] and we will evaluate it whereever we use it?